src/envoy/http/path_matcher/filter_fuzz_test.cc

```diff
@@ -0,0 +1,73 @@
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/server/mocks.h"
+
+#include "src/envoy/http/path_matcher/filter.h"
+#include "tests/fuzz/structured_inputs/path_matcher_filter.pb.validate.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace path_matcher {
+namespace fuzz {
+
+// Needed for logger macro expansion.
+namespace Logger = Envoy::Logger;
+
+void doTest(Filter& filter,
+            const espv2::tests::fuzz::protos::PathMatcherFilterInput& input) {
+  // Generate the user request.
+  auto headers =
+      Envoy::Fuzz::fromHeaders<Envoy::Http::TestRequestHeaderMapImpl>(
+          input.downstream_request().headers());
+
+  // Functions under test.
+  filter.decodeHeaders(headers, false);
+}
+
+DEFINE_PROTO_FUZZER(
+    const espv2::tests::fuzz::protos::PathMatcherFilterInput& input) {
+  ENVOY_LOG_MISC(trace, "{}", input.DebugString());
+
+  try {
+    Envoy::TestUtility::validate(input);
+
+    if (input.config().rules_size() < 1) {
+      throw Envoy::ProtoValidationException(
+          "At least 1 path matcher rule needed", input);
+    }
+  } catch (const Envoy::ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "Controlled proto validation failure: {}", e.what());
+    return;
+  }
+
+  // Setup mocks.
+  NiceMock<Envoy::Http::MockStreamDecoderFilterCallbacks>
+      mock_decoder_callbacks;
+  NiceMock<Envoy::Server::Configuration::MockFactoryContext>
+      mock_factory_context;
+
+  try {
+    // Create the filter.
+    FilterConfigSharedPtr config = std::make_shared<FilterConfig>(
+        input.config(), "fuzz-test-stats", mock_factory_context);
+    Filter filter(config);
+    filter.setDecoderFilterCallbacks(mock_decoder_callbacks);
```

Comment:
can we combine two try-catch blocks into one and not need to use ASSERT_NO_THROW?
the code should be much shorter

Reply:
Could you clarify how they can be combined? They check for different exceptions.

Also, we should ensure that `doTest` will not throw any exception. If we move it outside the try/catch block, then we need to make `Filter filter` a pointer in the test.

Reply:
If we don't care about exception type, we could do this:

filter_share_ptr filter
try {
     validdate proto
     create config and filter 
} catch (...)

doTest(filter, input)

Reply:
doTest() may not need to be a separate function

Reply:
But then even the mocks (`MockFactoryContext`) need to be moved up. Unless I make all of these pointers as well, this will result in a performance hit. The mocks will be constructed even when the proto is invalid.

What do you think is cleaner? I prefer the current solution, but I can change everything to pointers if needed.

Reply:
Though you don't need pointer to combine these two tryCatch if adding multiple catch block, I also prefer current struct, which looks clear.

Reply:
OK, it could be either way.   Current way is fine